src/util/Hooks.js

```diff
@@ -33,9 +33,9 @@ export async function executeHooksAfter(hook) {
 export function mergeHook(hooks, formName, hookToMerge) {
   const existingHook = hooks[formName] = hooks[formName] || { before: [], after: [] };
```

Comment:
Is this all still necessary if we have the default value setting (i.e. `|| []`) below?

Reply:
I think `hooks[formName]` could contain other hooks other than `before` or `after`?  Which would make this not initialize with default `{ before: [], after: [] };`

Reply:
ignore my last comment, it does look like this is not necessary with the default value setting below

Reply:
It's necessary.
I've found a bug in Gen 2 when added new tests for hooks.
If we have only "after" hook in options, and then trying to add new "before" hook with `.before()` API, we'll get `TypeError: Cannot read property 'concat' of undefined`

https://github.com/okta/okta-signin-widget/blob/bc61d536e2aece3e6e1afe2741d25e47295e301f/.widgetrc.sample.js#L20-L25

https://github.com/okta/okta-signin-widget/blob/36fc36650f9a229f0e2ff556c0fea63cead245fc/test/testcafe/spec/OktaSignIn_spec.js#L144-L146